draft-dt-tls-external-psk-guidance.md

```diff
@@ -292,6 +256,19 @@ pair-wise shared keys to achieve this. As another example, some systems require
 application-specific information in either PSKs or their identities. Identities may sometimes need to be routable,
 as is currently under discussion for EAP-TLS-PSK.
 
+
+# Recommendations for External PSK Usage
+
+
+Applications MUST use external PSKs that adhere to the following requirements:
+
+1. Each PSK SHOULD be derived from at least 128 bits of entropy, MUST be at least 128 bits long, and SHOULD be combined with a DH exchange for forward secrecy. Low entropy PSKs, i.e., those derived from less than 128 bits of entropy, MUST be combined with a Password Authenticated Key Exchange (PAKE) mechanism.
+2. Each PSK MUST NOT be shared between with more than two logical nodes. As a result, an agent that acts as both a client and a server MUST use distinct PSKs when acting as the client from when it is acting as the server. 
+3. Nodes SHOULD use external PSK importers {{!I-D.ietf-tls-external-psk-importer}} when configuring PSKs for a pair of TLS client and server.
+4. Where possible the master PSK (that which is fed into the importer) SHOULD be deleted after the imported keys have been generated. This protects an attacker from bootstrapping a compromise of one node into the ability to attack connections between any node; otherwise the attacker can recover the master key and then re-run the importer itself.
+
+ {{!RFC8446}} mandates that each external PSK can only be used with a single hash function. This was done to simplify the protocol analysis. Using {{!I-D.ietf-tls-external-psk-importer}} can provide an additional benefit as it produces a set of candidate PSKs, each of which are bound to a specific KDF and hash function.
```

Comment:
This paragraph does not add information. Please remove it.

Reply:
With Björn's agreement, this is now removed.